Sources/Signal.swift

```diff
@@ -426,6 +426,12 @@ extension Signal: SignalProtocol {
 	}
 }
 
+extension Signal: SignalProducerRepresentable {
+	public var producer: SignalProducer<Value, Error> {
+		return SignalProducer(self)
```

Comment:
Doesn't this mean taking some performance hit? The `SignalProducer` adds extra work on top of the `Signal`.